yt/visualization/base_plot_types.py

```diff
@@ -276,33 +276,26 @@ def _init_image(self, data, cbnorm, cblinthresh, cmap, extent, aspect):
             transform = self.axes.transData
         else:
             transform = self._transform
+
         if hasattr(self.axes, "set_extent"):
-            # CartoPy hangs if we do not set_extent before imshow if we are
-            # displaying a small subset of the globe.  What I believe happens is
-            # that the transform for the points on the outside results in
-            # infinities, and then the scipy.spatial cKDTree hangs trying to
-            # identify nearest points.
-            #
-            # Also, set_extent is defined by cartopy, so not all axes will have
-            # it as a method.
-            #
-            # A potential downside is that other images may change, but I believe
-            # the result of imshow is to set_extent *regardless*.  This just
-            # changes the order in which it happens.
-            #
-            # NOTE: This is currently commented out because it breaks in some
-            # instances.  It is left as a historical note because we will
-            # eventually need some form of it.
-            # self.axes.set_extent(extent)
-
-            # possibly related issue (operation order dependency)
-            # https://github.com/SciTools/cartopy/issues/1468
-
-            # in cartopy 0.19 (or 0.20), some intented behaviour changes produced an
-            # incompatibility here where default values for extent would lead to a crash.
-            # A solution is to let the transform object set the image extents internally
-            # see https://github.com/SciTools/cartopy/issues/1955
-            extent = None
+            # some projections have trouble when passing extents at or near the
+            # limits. So we only set_extent when the plot is a subset of the
+            # globe, within the tolerance of the transform.
+
+            # add validation to make sure transform IS a cartopy crs? pretty sure
+            # it should always be if we end up here....
+
+            # note that `set_extent` here is setting the extent of the axes.
+            # still need to pass the extent arg to imshow below in order to
+            # ensure that it is properly scaled. also note that set_extent
+            # expects values in the coordinates of the transform: it will
+            # calculate the coordinates in the projection.
+            global_extent = transform.x_limits + transform.y_limits
+            thresh = transform.threshold
+            if all([extent[ie] < (global_extent[ie] - thresh) for ie in range(4)]):
```

Comment:
Quick optimisation tip here: `all` accepts a generator, so exhaustive consumption of the iterable via conversion into a list  isn't needed
```suggestion
            if all(extent[ie] < (global_extent[ie] - thresh) for ie in range(4)):
```